Prox/Prox/PlaceDetails/PlaceDetailsCardView.swift

```diff
@@ -186,19 +188,57 @@ class PlaceDetailsCardView: UIScrollView {
         if frame != newFrame { frame = newFrame }
     }
 
+    // TODO: don't duplicate PlaceCarousel code
+    func updateUI(forPlace place: Place) {
+        // Labels will gracefully collapse on nil.
+        titleLabel.text = place.name
+        categoryLabel.text = place.categories?.prefix(MaxDisplayedCategories).joined(separator: " • ")
+        urlLabel.text = place.url ?? nil
+
+        // TODO: error states for missing data.
+        updateHoursUI(place.hours)
+
+        updateReviewUI(fromProvider: place.yelpProvider, onView: yelpReviewView)
+        updateReviewUI(fromProvider: place.tripAdvisorProvider, onView: tripAdvisorReviewView)
+    }
+
+    private func updateReviewUI(fromProvider provider: ReviewProvider?, onView view: ReviewContainerView) {
+        guard let provider = provider else {
+            view.score = 0
+            view.numberOfReviewersLabel.text = "No data found" // TODO: error state.
+            return
+        }
+
+        // TODO: error states
+        view.score = provider.rating ?? 0
+        view.numberOfReviewersLabel.text = "\(provider.totalReviewCount ?? 0) Reviews"
+    }
+
+    private func updateHoursUI(_ hours: [DayOfWeek:OpenHours]?) {
```

Comment:
Again, this kind of logic really should be in the `PlaceDetailsCardViewController`, but I can learn to live with it :)